users/userRouter.js

```diff
@@ -1,47 +1,92 @@
-const express = 'express';
+const express = require('express');
 
 const router = express.Router();
 
-router.post('/', (req, res) => {
+const userDb = require('./userDb.js')
+const postDb = require('../posts/postDb.js')
 
-});
+router.use(express.json())
 
-router.post('/:id/posts', (req, res) => {
+router.post('/', validateUser, (req, res) => {
+  const user = req.body
+  userDb.insert(user)
+    .then(data => res.status(201).json(user))
+    .catch(err => res.status(500).json({ message: "something has gone wrong" }))
+});
 
+router.post('/:id/posts', validateUserId, validatePost, (req, res) => {
+  const id = req.params.id
+  const post = {
+    "user_id": id,
+    "text": req.body.text
+  }
+  postDb.insert(post)
+    .then(data => res.status(201).json(data))
+    .catch(err => res.status(500).json({ message: "something has gone wrong" }))
 });
 
 router.get('/', (req, res) => {
-
+  userDb.get()
+    .then(data => res.status(200).json(data))
+    .catch(err => res.status(500).json({ message: "something has gone wrong" }))
 });
 
-router.get('/:id', (req, res) => {
-
+router.get('/:id', validateUserId, (req, res) => {
+  const id = req.params.id
+  userDb.getById(id)
+    .then(data => res.status(200).json(data))
+    .catch(err => res.status(500).json({ message: "something has gone wrong" }))
 });
 
-router.get('/:id/posts', (req, res) => {
-
+router.get('/:id/posts', validateUserId, (req, res) => {
+  userDb.getUserPosts(userId)
+    .then(data => res.status(200).json(data))
+    .catch(err => res.status(500).json({ message: "something has gone wrong" }))
 });
 
-router.delete('/:id', (req, res) => {
-
+router.delete('/:id', validateUserId, (req, res) => {
+  const id = req.params.id
+  userDb.remove(id)
+    .then(data => res.status(204).json({ message: "user deleted" }))
+    .catch(err => res.status(500).json({ message: "something has gone wrong" }))
 });
 
-router.put('/:id', (req, res) => {
-
+router.put('/:id', validateUserId, (req, res) => {
+  const id = req.params.id
+  const changes = req.body
+  userDb.update(id, changes)
+    .then(data => res.status(200).json({ message: "user updated" }))
+    .catch(err => res.status(500).json({ message: "something has gone wrong" }))
 });
 
 //custom middleware
 
 function validateUserId(req, res, next) {
-
+  const id = req.params.id
+  userDb.getById(id)
+    .then(data => data ? req.user = data : res.status(400).json({ message: "invalid user id" }))
+    .catch()
+  next();
 };
 
 function validateUser(req, res, next) {
-
+  if (!req.body) {
+    res.status(400).json({ message: "missing user data" })
+  }
+  if (!req.body.name) {
+    res.status(400).json({ message: "missing required name field" })
+  }
+  next();
 };
 
 function validatePost(req, res, next) {
-
+  if (!req.body) {
+    res.status(400).json({ message: "missing post data" })
```

Comment:
!req.body isn't going to be very useful, even if nothing is sent with the request req.body will still be an empty object. req.body.text is all we really need to check for